packages/core/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -233,16 +229,19 @@ async function validateJavaHome() {
 
 export async function startTransformByQ() {
     let intervalId = undefined
+    let userInputState = undefined
 
-    // Validate inputs. If failed, Error will be thrown and execution stops
-    const userInputState = await validateTransformationJob()
-
-    await setMaven()
-
-    await validateJavaHome()
+    try {
+        // Validate inputs. If failed, Error will be thrown and execution stops
+        userInputState = await validateTransformationJob()
+        await setMaven()
+        await validateJavaHome()
+    } catch (error: any) {
+        throw new ToolkitError('', { code: 'InputValidationFailed' })
```

Comment:
Fixed error handling so that we always use regular `Error` (`ToolkitError` not needed since we manually show our own/custom notifications) -- but this means users get a notification saying "Could not execute command aws.q.transform" when `Error` is thrown -- so I handle this by catching all of those and using a regular `ToolkitError` just once (here) with an empty string, rather than using `ToolkitError` with an empty string in 10+ places.

Reply:
This seems like hacks upon hacks.

> since we manually show our own/custom notifications)

Why?

Reply:
I wouldn't say hack necessarily but more so just that I've been doing: showing notifications via `vscode.window.showError/InfoMessage` + throwing regular `Error` versus bundling the notification with the `ToolkitError`. We previously had both types of errors throughout which was confusing at times, so wanted to pick one and stick with it.

Also, there are some places such as https://github.com/aws/aws-toolkit-vscode/blob/cf3f0fe020370ed80f55c7f4aba8f3691e8ce73c/packages/core/src/codewhisperer/commands/startTransformByQ.ts#L566 where we allow the user to click a "Submit feedback" button through the notification, which I don't believe I can implement with the `ToolkitError`. So from this for consistency I figured it's better to use `vscode.window` to do notification handling everywhere, and would like to avoid duplicate notifications being shown, hence why I got rid of the many `ToolkitError`s.

Reply:
Throwing an error with an empty message could lead to some strange situations and just isn't an expected or recommended practice. That's not done anywhere else in the codebase (except a few tests).

The fact that some accidental situations have lead to this workaround doesn't change the fact that there is a design issue here.